test/bluebird.test.js

```diff
@@ -94,10 +100,189 @@ describe('bluebird', function () {
+    it('should continue retry while error is instanceof match', function () {
+        return expect(retry(function () {
+        count++;
+
+        if (count < 4) {
+          return Promise.reject(new Error(soRejected));
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: Error
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(4);
+      });
+    });
+    
+    it('should continue retry while error is equal to match string in array', function () {
+        return expect(retry(function () {
+        count++;
+
+        if (count < 4) {
+          return Promise.reject(soRejected);
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: [soRejected + 1, soRejected]
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(4);
+      });
+    });
+
+    it('should reject immediately if error is not equal to match string in array', function () {
+        return expect(retry(function () {
+        count++;
+
+        return Promise.reject(soRejected);
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: [soRejected + 1, soRejected + 2]
+      })).to.eventually.be.rejectedWith(soRejected).then(function () {
+        expect(count).to.equal(1);
+      });
+    });
+
+    it('should reject immediately if error is not instanceof match in array', function () {
+        return expect(retry(function () {
+        count++;
+
+        return Promise.reject(new Error(soRejected));
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: [soRejected + 1, function foo(){}],
+      })).to.eventually.be.rejectedWith(Error).then(function () {
+        expect(count).to.equal(1);
+      });
+    });
+
+    it('should continue retry while error is instanceof match in array', function () {
+        return expect(retry(function () {
+        count++;
+
+        if (count < 4) {
+          return Promise.reject(new Error(soRejected));
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15,
+        stepbackBase: 1,
+        match: [soRejected + 1, Error]
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(4);
+      });
+    });
+  });
+
+  describe('stepback', function () {
+    it('should resolve after 10 retries and an eventual delay over 1.2 seconds using default stepback', function () {
+        var startTime = moment();
+        return expect(retry(function () {
+        count++;
+
+        if (count < 10) {
+          return Promise.reject(soRejected);
+        }
+
+        return Promise.resolve(soResolved);
+      }, {
+        max: 15
+      })).to.eventually.be.equal(soResolved).then(function () {
+        expect(count).to.equal(10);
+        expect(moment().diff(startTime)).to.be.above(1200);
+      });
+    });
```

Comment:
Would also be easier to test with a stub on Promise.delay and just verifying that the call values to it are as expected